### PR TITLE
libvmi.WithMasqueradeNetworking(): return a single option instead of a slice

### PR DESCRIPTION
**Pull request by iholder101**
### What this PR does
Before this PR:
`libvmi.WithMasqueradeNetworking()` was returning a slice of options. This is inconsistent with how libvmi is usually used, and enforces the user to create a slice than expand it which makes the code ugly for unnecessary reasons.

As an example, this is how calling it looked like:
```go 
			vmi := libvmifact.NewAlpineWithTestTooling(
				append(libnet.WithMasqueradeNetworking(), libvmi.WithCPUCount(1, 2, 1))...,
			)
```

After this PR:
`libvmi.WithMasqueradeNetworking()` returns a signle option like any other libvmi function. The code will look as following:
```go 
			vmi := libvmifact.NewAlpineWithTestTooling(
				libnet.WithMasqueradeNetworking(), libvmi.WithCPUCount(1, 2, 1)),
			)
```

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [X] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [X] PR: The PR description is expressive enough and will help future contributors
- [X] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [X] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] ~Upgrade: Impact of this change on upgrade flows was considered and addressed if required~
- [X] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] ~Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.~
- [ ] ~Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered~

### Release note

```release-note
NONE
```

### CODE REVIEW

File: tests/network/probes.go
Comment (fossedihelm) [inline tests/network/probes.go:379]: same as above

**Review by fossedihelm — CHANGES_REQUESTED** [review]
Thank you @iholder101! Great initiative! I only request 2 changes

**Comment by iholder101** [comment]
> Thank you @iholder101! Great initiative! I only request 2 changes

Nice catch! Thanks @fossedihelm, PTAL

**Review by fossedihelm — APPROVED** [review]
Thank you! looks good to me

**Comment by iholder101** [comment]
Hi again @fossedihelm!

I've missed some more usages that were added during the rebase, now they're all addressed :)

Also, `pull-kubevirt-unit-test-arm64`'s failure is unrelated to this PR. It fails over `unable to manipulate /tmp's contents, exiting` which occurs on other PRs as well (e.g. [here](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/12090/pull-kubevirt-unit-test-arm64/1803332745649197056))

**Comment by fossedihelm** [comment]
Sorry, missed the latest push.
/lgtm

File: tests/network/probes.go
Comment (EdDev) [inline tests/network/probes.go:379]: I think the reason `withMasqueradeNetworkingAndFurtherUserConfig` existed is gone now and you can drop this helper.

**Review by EdDev — COMMENTED** [review]
Nice cleanup, thanks!

It will be great to see the reasoning from the PR description added to the commit.
Also, it does help (me) when the subject line has a common format like this: `topic: subject` (e.g. `libvmi: ...`. Then I have context to understand what changed and where by just looking at the git log oneliner.

BTW: It may be opinionated, but I do not understand the added value of the current commit split. I usually split to tell a "story" and help the reviewer in going over the changes. But in this case even the reasoning is common.

File: tests/network/probes.go
Comment (iholder101) [reply tests/network/probes.go:379]: What do you mean? It's still used in a bunch of places (e.g. [here](https://github.com/kubevirt/kubevirt/blob/dcc0fcb5096d5bb232ee70d3c113c8d300552487/tests/network/probes.go#L134)).
TBH I'm not sure what's the reason it existed in the first place. Can we leave this to a follow-up?

Comment (EdDev) [reply tests/network/probes.go:379]: Yes, it is used in several places and now it can be removed and the options accumulated directly without this helper.

I won't insist, but IMO it is a fit in this PR.

**Review by EdDev — APPROVED** [review]
From libvmi point of view, this looks great, thank you.

/approve

For the callers code, you will need to ask some other approver or self-approve.

File: tests/network/probes.go
Comment (iholder101) [reply tests/network/probes.go]: Removed the function :) Thanks!

**Review by EdDev — COMMENTED** [review]
Thanks!

/approve

**Review by EdDev — APPROVED** [review]

**Comment by EdDev** [comment]
> This pull-request has been approved by: _[EdDev](https://github.com/kubevirt/kubevirt/pull/12135#pullrequestreview-2129317586)_
> **Once this PR has been reviewed and has the lgtm label**, please ask for approval from [iholder101](https://github.com/iholder101). For more

The bot asked to ask you :)

But it seems you need to rebase now :/

**Comment by iholder101** [comment]
> > This pull-request has been approved by: _[EdDev](https://github.com/kubevirt/kubevirt/pull/12135#pullrequestreview-2129317586)_
> > **Once this PR has been reviewed and has the lgtm label**, please ask for approval from [iholder101](https://github.com/iholder101). For more
> 
> The bot asked to ask you :)

:laughing: 

> But it seems you need to rebase now :/

Yeah it'll be challenging to merge this due to rebases..
Let's hope some other reviewer can look at this soon enough

**Comment by enp0s3** [comment]
/approve
@iholder101 Thank you!

**Comment by fossedihelm** [comment]
/lgtm
Let's try to reduce the conflicts

File: tests/hotplug/memory.go
Comment (acardace) [inline tests/hotplug/memory.go:63]: This is not correct, `opts` are not added anymore.

Comment (fossedihelm) [reply tests/hotplug/memory.go:63]: :man_facepalming:

Comment (EdDev) [reply tests/hotplug/memory.go:63]: Indeed missed, but it is also worrying that it passed e2e tests.
If it passed the tests, what does it mean about the [test](https://github.com/kubevirt/kubevirt/blob/0145200b4213460164d1603cddb62d9bcd068d5a/tests/hotplug/memory.go#L149) that uses it?

Comment (acardace) [reply tests/hotplug/memory.go:63]: Unfortunately our CI does not run any presubmit lane with hugepages support at the moment, so bad luck.

Comment (xpivarc) [reply tests/hotplug/memory.go:63]: @dhiller @brianmcarey can we fix this?

Comment (EdDev) [reply tests/hotplug/memory.go:63]: > @dhiller @brianmcarey can we fix this?

If we can run it as part of the current lanes (i.e. change the nodes to have hugepages), great.
But I am not in favor of running another lane as a presubmit.

Do we have a postsubmit job that runs this test? Has it popped up as an alarm that it failed?

Comment (acardace) [reply tests/hotplug/memory.go:63]: Yes, it's fixed already https://github.com/kubevirt/kubevirt/commit/f8915253592692ff8a8f7c48b5b9f2b10b731d2f

Comment (acardace) [reply tests/hotplug/memory.go:63]: I think we should also add a linter for unused function parameters.

**Pull request merged**
